gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DataSource-pouchDB.js

```diff
@@ -0,0 +1,194 @@
+/*!
+Copyright 2016 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+/* eslint-env browser */
+/* eslint strict: ["error", "function"] */
+
+var fluid = fluid || require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+(function () {
+
+    "use strict";
+
+    // This pouchDB data source is to support the in-browser testing that uses pouchDB as the backend data storage.
+    // It overrides default setImpl() and getImpl() provided by "kettle.dataSource.URL" with gpii.pouch API calls.
+    // Note: This component requires an instance of gpii.pouch (see https://github.com/the-t-in-rtf/gpii-pouchdb/blob/GPII-1897/docs/pouchdb.md)
+    // to be instantiated beforehand.
+    fluid.defaults("gpii.dataSource.pouchDB", {
+        gradeNames: ["kettle.dataSource.URL"],
```

Comment:
I've improved the test for `gpii.dataSource.pouchDB` to use `writable: true` option hooking up the writable grade instead of using the writable grade itself directly. Yes, the `readOnlyGrade` is required to be in place at using `writable: true`. Also removed the linkage grade which doesn't seem to be needed.